src/main/java/org/alfresco/repo/copy/CopyServiceImpl.java

```diff
@@ -1519,4 +1517,25 @@ public String getTopLevelNodeNewName(NodeRef sourceNodeRef, NodeRef targetParent
         return null;
     }
 
+    /**
+     * Builds name by appending copy label for each copy level.
+     */
+    String buildNewName(NodeRef destinationParent, QName assocTypeQName, String newName)
+    {
+        while (this.internalNodeService.getChildByName(destinationParent, assocTypeQName, newName) != null)

```

Comment:
Does the while loop need to be part of this method?
Removing the while loop would simplify the method and also make the unit tests simpler.